### PR TITLE
ui: add error handling and form validation to job create template

### LINKED ISSUE #15766: Handle error messages on Create Job Template form

### PR DESCRIPTION
**Pull request by ChaiWithJai**
Closes #15766

### CODE REVIEW

**Comment by ChaiWithJai** [comment]
An unrelated test is failing because `pretender` isn't running an `assertion`. Unsure why this is the case here.

File: ui/app/controllers/jobs/run/templates/new.js [hunk 1]
```diff
@@ -19,6 +22,17 @@ export default class JobsRunTemplatesController extends Controller {
     return namespaces;
   }
 
+  get isDuplicateTemplate() {
+    const templates = this.store.peekAll('variable');
```
Comment (philrenaud) [inline ui/app/controllers/jobs/run/templates/new.js:26]: Suggestion: either use the queried variables from the route's model at jobs/run/templates/new, or else alias this somewhere outside the isDuplicateTemplate() getter. As it stands, this is going to make a store.peekAll() request every time you type a character into the path input element, which is probably doing more work than it needs to.

**Review by philrenaud — APPROVED** [review]
Just a small suggestion, otherwise this seems like it works well. test-ui seems to be failing in a couple spots; once that's good, this seems good to merge.

File: ui/app/controllers/jobs/run/templates/new.js (hunk 1 above)
Comment (ChaiWithJai) [reply ui/app/controllers/jobs/run/templates/new.js:26]: Great callout! I'll address this in the `edit` template PR!

**Pull request merged**